go/proof.go

```diff
@@ -58,6 +58,7 @@ var SmtSpec = &ProofSpec{
 		Hash:            HashOp_SHA256,
 	},
 	MaxDepth: 256,
+  PrehashComparedKey: HashOp_SHA256,
```

Comment:
I find the naming between `PrehashKey` and `PrehashComparedKey` confusing. Thanks @AdityaSripal for helping me walkthrough the difference. From my understanding:

Fields in the `LeafOp` represent how we encode the Leaf node when constructing and proving proofs. The `PrehashComparedKey` is used to construct the key used in the tree (ie its location). 

Are there ways we could improve the field naming here such that its clear that these fields are distinctly different despite both being prehashes on a key. In my terminology, I see `PrehashComparedKey` as a prehash on the preimage of the key, while `PrehashKey` is a prehash of the key. Maybe something like `PrehashPreimageKey`?